vms/proposervm/proposer/windower.go

```diff
@@ -51,6 +62,23 @@ type Windower interface {
 		validatorID ids.NodeID,
 		maxWindows int,
 	) (time.Duration, error)
+
+	ExpectedProposer(
+		ctx context.Context,
+		chainHeight,
+		pChainHeight uint64,
+		parentBlockTime,
+		blockTime time.Time,
+	) (ids.NodeID, error)
+
+	MinDelayForProposer(
+		ctx context.Context,
+		chainHeight,
+		pChainHeight uint64,
+		parentBlockTime time.Time,
+		nodeID ids.NodeID,
+		startTime time.Time,
+	) (time.Time, error)
```

Comment:
Can we add some tests for this? I think that the returned `time.Time` should be:
1. In the range `[startTime, infinity)`.
2. If possible (i.e. unless we are returning `startTime`) at the start of a slot.